src/client/Client.cc

```diff
@@ -8591,8 +8590,6 @@ int Client::_lookup_parent(Inode *ino, const UserPerm& perms, Inode **parent)
       *parent = target.get();
       _ll_get(*parent);
       ldout(cct, 8) << __func__ << " found parent " << (*parent)->ino << dendl;
-    } else {
-      *parent = NULL;
```

Comment:
The rest of the PR looks fine but I generally prefer that we *do* set these values unconditionally. We've had a number of bugs not doing so in other parts of the code and I think we try to initialize them everywhere now.

Reply:
I prefer leaving that sort of initialization to the caller, but I'll just drop that patch, as it's not critical here anyway.